scene/main/viewport.cpp

```diff
@@ -89,20 +93,17 @@ void ViewportTexture::set_viewport_path_in_scene(const NodePath &p_path) {
 	}
 
 	path = p_path;
+	vp_changed = true;
 
 	if (vp) {
 		vp->viewport_textures.erase(this);
 		vp = nullptr;
 	}
 
-	if (proxy_ph.is_valid()) {
-		RS::get_singleton()->free(proxy_ph);
-	}
-	if (proxy.is_valid()) {
-		RS::get_singleton()->free(proxy);
+	if (proxy.is_valid() && proxy_ph.is_null()) {
+		proxy_ph = RS::get_singleton()->texture_2d_placeholder_create();
+		RS::get_singleton()->texture_proxy_update(proxy, proxy_ph);
 	}
-	proxy_ph = RID();
-	proxy = RID();
```

Comment:
Wouldn't it be enough to just remove this?

Aside, it seems that the original intent is that `proxy` and `proxy_ph` are both either set or null and the changes in this PR seem to challenge that.

Reply:
What is considered in `_setup_local_to_scene()` is actually the case of setting the target viewport for the first time. The case of changing the target viewport is actually not considered.

The main consideration here is when the `viewport_path` is reset in the editor. 

`proxy` may be valid when reset. Additionally, `viewport_path` has the potential to be reset to an empty path, which means that `_set_viewport_path_in_scene()` will not be called, and it should disconnect from the previous target viewport, this needs to be done in `set_viewport_path_in_scene()`.

In other words, `set_viewport_path_in_scene()` should reset the state, making the `ViewportTexture` look like it was never set.

Reply:
I think I get it now. Thanks for the explanation.

Last thing I would ask is maybe one/some comment(s) in the code, maybe describing what `vp_changed` is for.